package/android/src/main/java/com/mrousavy/camera/core/CameraDeviceDetails.kt

```diff
@@ -95,27 +95,24 @@ class CameraDeviceDetails(private val cameraManager: CameraManager, private val
   private val size35mm = Size(36, 24)
 
   private fun getDeviceTypes(): ReadableArray {
-    // TODO: Check if getDeviceType() works correctly, even for logical multi-cameras
-
     // To get valid focal length standards we have to upscale to the 35mm measurement (film standard)
     val cropFactor = size35mm.bigger / sensorSize.bigger
 
     val deviceTypes = Arguments.createArray()
 
-    // https://en.wikipedia.org/wiki/Telephoto_lens
-    val containsTelephoto = focalLengths.any { l -> (l * cropFactor) > 35 } // TODO: Telephoto lenses are > 85mm, but we don't have anything between that range..
-    // val containsNormalLens = focalLengths.any { l -> (l * cropFactor) > 35 && (l * cropFactor) <= 55 }
-    // https://en.wikipedia.org/wiki/Wide-angle_lens
-    val containsWideAngle = focalLengths.any { l -> (l * cropFactor) >= 24 && (l * cropFactor) <= 35 }
-    // https://en.wikipedia.org/wiki/Ultra_wide_angle_lens
-    val containsUltraWideAngle = focalLengths.any { l -> (l * cropFactor) < 24 }
-
-    if (containsTelephoto)
-      deviceTypes.pushString("telephoto-camera")
-    if (containsWideAngle)
-      deviceTypes.pushString("wide-angle-camera")
-    if (containsUltraWideAngle)
-      deviceTypes.pushString("ultra-wide-angle-camera")
+    focalLengths.forEach { focalLength ->
+      // scale to the 35mm film standard
+      val l = focalLength * cropFactor
+      when {
+        // https://en.wikipedia.org/wiki/Ultra_wide_angle_lens
+        l < 24f -> deviceTypes.pushString("ultra-wide-angle-camera")
+        // https://en.wikipedia.org/wiki/Wide-angle_lens
+        l in 24f..43f -> deviceTypes.pushString("wide-angle-camera")
+        // https://en.wikipedia.org/wiki/Telephoto_lens
+        l > 43f -> deviceTypes.pushString("telephoto-camera")
```

Comment:
Can `focalLengths` contain more than one element? if it can, you may end up pushing repeated types to  `deviceTypes`. The old code handled this better I think.

Reply:
Here's the thing; technically there could be a single camera device that had two similar cameras (eg two wide angle cameras; for example one for depth and one for color).

With the old code, that resulted in [wide-angle] (1x), with the new code that results in [wide-angle, wide-angle] (2x), being more accurate.

Do you think that's confusing?

Reply:
Wondering if it qualifies as "breaking changes".